src/main/java/com/elastic/support/ElasticClientService.java

```diff
@@ -0,0 +1,78 @@
+package com.elastic.support;
+
+import com.elastic.support.config.BaseConfig;
+import com.elastic.support.config.ElasticClientInputs;
+import com.elastic.support.rest.RestClient;
+import com.elastic.support.rest.RestClientBuilder;
+import com.elastic.support.util.SystemProperties;
+import com.elastic.support.util.SystemUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class ElasticClientService extends BaseService {
+
+
+    protected RestClient createGenericClient(BaseConfig config, ElasticClientInputs inputs) {
+        RestClientBuilder builder = new RestClientBuilder();
+
+        return builder
```

Comment:
If you provided a private version of this method that returned the `RestClientBuilder`, then both of them could share the "generic" builder and `createEsRestClient` could just add what's unique to it.